src/main/kotlin/com/healthmetrix/labres/notifications/NotificationsConfig.kt

```diff
@@ -0,0 +1,62 @@
+package com.healthmetrix.labres.notifications
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.firebase.FirebaseApp
+import com.google.firebase.FirebaseOptions
+import com.google.firebase.messaging.FirebaseMessaging
+import com.healthmetrix.labres.logger
+import com.healthmetrix.labres.secrets.Secrets
+import java.io.ByteArrayInputStream
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+
+@Configuration
+class NotificationsConfig(
+    @Value("\${labres.stage}")
+    private val stage: String,
+    private val secrets: Secrets
+) {
+
+    @Bean
+    @Profile("notify")
+    fun httpNotifier(objectMapper: ObjectMapper): Notifier<Notification.HttpNotification> =
+        secrets.get("lab-res/$stage/notification/basic-auth")
+            ?.let { objectMapper.readValue(it, HttpNotificationConfig.BasicAuth::class.java) }
+            ?.let(::HttpNotificationConfig)
+            ?.let(::HttpNotifier)
+            ?: throw InternalError("Could not retrieve basic auth credentials for HTTP notification")
+
+    @Bean
+    @Profile("notify")
+    fun firebaseNotifier(): Notifier<Notification.FcmNotification> {
+        val credentials = secrets.get("lab-res/$stage/fcm/credentials")
+            ?.toByteArray(Charsets.UTF_8)
+            ?.let(::ByteArrayInputStream)
+            ?.let(GoogleCredentials::fromStream)
+            ?: throw InternalError("Could not retrieve FCM credentials")
+
+        val options = FirebaseOptions.builder()
+            .setCredentials(credentials)
+            .build()
+
+        val app = FirebaseApp.initializeApp(options)
+
+        val messaging = FirebaseMessaging.getInstance(app)
+
+        val dryRun = stage == "local"
```

Comment:
does it make sense to make the `dryRun` configurable from `application.yaml`?

Reply:
I'm trying to think of situations where we actually need this to be false besides the initial testing. We already have the `notify` profile in place to toggle the feature of sending notifications or just log them. If I'd add the dryRun flag to the `application.yaml` I would probably set it to true by default and only to false on the `notify` profile. But if this one isn't enabled, the code won't be added to the spring context anyways.
I also can't use the `FcmNotifier` always and toggle only the `dryRun` flag based on the `notify` profile, cause then the instantiation of the FcmNotifier would probably fail for not having a valid Google Cloud service account credentials JSON as secret. 
So long story short, I don't see too much value having the flag at all and not just hard coding it to always false. Would leave it like it is tho in case someone needs to ever change the notification stuff and wants a quite easy way for testing it first locally.

Reply:
alright